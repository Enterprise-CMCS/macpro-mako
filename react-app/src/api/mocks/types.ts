import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";

type GetTypesBody = { authorityId: number };
type GetSubTypesBody = { authorityId: number; typeIds: number[] };

export const handlers = [
  http.post<any, GetTypesBody>("/os/getTypes", async ({ request }) => {
    const { authorityId } = await request.json();

    if (authorityId === -1) {
      throw Error("useGetTypes > mockFetch: Expected error thrown by test.");
    }

    return HttpResponse.json({
      hits: {
        hits: mockData.filter(
          (item) => item._source.authorityId === authorityId && !item._source.typeId,
        ),
      },
    });
  }),

  http.post<any, GetSubTypesBody>("/os/getSubTypes", async ({ request }) => {
    const { authorityId, typeIds } = await request.json();

    if (authorityId === -1) {
      throw Error("useGetSubTypes > mockFetch: Expected error thrown by test.");
    }

    const filteredData = mockData.filter(
      (item) =>
        item._source.authorityId === authorityId &&
        typeIds.some((typeId) => item._source.typeId === typeId),
    );

    return HttpResponse.json({
      hits: {
        hits: filteredData,
      },
    });
  }),
];

const mockData = [
  { _source: { id: 101, authorityId: 1, name: "typeOne" } },
  { _source: { id: 102, authorityId: 1, name: "typetwo" } },
  { _source: { id: 103, authorityId: 2, name: "typethree" } },
  { _source: { id: 101, authorityId: 1, name: "subtypeOne", typeId: 1 } },
  { _source: { id: 102, authorityId: 1, name: "subtypetwo", typeId: 2 } },
  { _source: { id: 103, authorityId: 2, name: "subtypethree", typeId: 1 } },
  { _source: { id: 104, authorityId: 2, name: "subtypethree", typeId: 4 } },
  { _source: { id: 105, authorityId: 2, name: "subtypethree", typeId: 3 } },
];

export const server = setupServer(...handlers);
