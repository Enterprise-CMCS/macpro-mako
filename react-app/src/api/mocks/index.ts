export * as mockSubmit from "./submit";
export * as mockTypes from "./types";
