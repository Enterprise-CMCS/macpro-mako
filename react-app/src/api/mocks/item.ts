import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";
import { opensearch, SEATOOL_STATUS } from "shared-types";

type GetItemBody = { id: string };
type ItemTestFields = Pick<
  opensearch.main.Document,
  "id" | "seatoolStatus" | "actionType"
>;

const ID_SEPARATOR = "-";
type IdParamKey = keyof opensearch.main.Document;
// Because getItem(id: string) doesn't allow for easy object mocking,
// to make tests easier, you can add params to the ids your tests use
// and mock specific attributes.
// e.x. existing-approved-actionType=New
const getIdParam = (id: string, key: IdParamKey) =>
  id
    .split(ID_SEPARATOR)
    .find((param: string) => param.includes(key))
    ?.slice(key.length + 1); // + 1 to cover the `=` sign

const handlers = [
  http.post<GetItemBody, GetItemBody>(
    "/item-mock-server",
    async ({ request }) => {
      const { id } = await request.json();
      return id.includes("existing")
        ? HttpResponse.json({
            _id: id,
            _source: {
              id: id,
              seatoolStatus: id.includes("approved")
                ? SEATOOL_STATUS.APPROVED
                : SEATOOL_STATUS.PENDING,
              actionType: getIdParam(id, "actionType") || "New",
            } satisfies ItemTestFields,
          })
        : new HttpResponse(null, { status: 404 });
    },
  ),
  http.post<GetItemBody, GetItemBody>(
    "/item-mock-server-bad",
    async ({ request }) => {
      const { id } = await request.json();
      return id.includes("existing")
        ? HttpResponse.json({
            _id: id,
            _source: {
              id: id,
              seatoolStatus: id.includes("approved")
                ? SEATOOL_STATUS.APPROVED
                : SEATOOL_STATUS.PENDING,
              actionType: getIdParam(id, "actionType") || "New",
            } satisfies ItemTestFields,
          })
        : new HttpResponse(null, { status: 404 });
    },
  ),
];

export const server = setupServer(...handlers);
