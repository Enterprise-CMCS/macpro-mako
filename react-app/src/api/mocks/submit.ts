import { http, HttpResponse } from "msw";
import { setupServer } from "msw/node";

const handlers = [
  http.put("/upload", async () => {
    return new HttpResponse(null, { status: 200 });
  }),
  http.post("/getUploadUrl", () => {
    return HttpResponse.json(
      {
        url: "/upload",
        key: "test-key",
        bucket: "test-bucket",
      },
      { status: 200 },
    );
  }),
  http.post("/test", () => {
    return HttpResponse.json({ message: "pass" }, { status: 200 });
  }),
];

export const server = setupServer(...handlers);
