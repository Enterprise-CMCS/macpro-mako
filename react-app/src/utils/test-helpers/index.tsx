export * from "./dashboard";
export * from "./render";
export * from "./renderForm";
export * from "./skipCleanup";
export * from "./uploadFiles";
