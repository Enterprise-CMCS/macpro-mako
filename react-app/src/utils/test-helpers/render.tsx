import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import React, { ReactElement } from "react";
import { createMemoryRouter, MemoryRouter, RouterProvider } from "react-router";
import { render } from "@testing-library/react";

export const createTestQueryClient = () =>
  new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
    logger: {
      log: console.log,
      warn: console.warn,
      error: () => {},
    },
  });

export const queryClientWrapper = ({ children }: { children: ReactElement }) => (
  <QueryClientProvider client={createTestQueryClient()}>{children}</QueryClientProvider>
);

export const memoryRouterWrapper = ({ children }: { children: ReactElement }) => (
  <MemoryRouter>{children}</MemoryRouter>
);

export const renderWithQueryClient = (element: ReactElement) =>
  render(element, {
    wrapper: ({ children }) => (
      <QueryClientProvider client={createTestQueryClient()}>
        <MemoryRouter>{children}</MemoryRouter>,
      </QueryClientProvider>
    ),
  });

export const renderWithMemoryRouter = (
  element: ReactElement,
  ...routing: Parameters<typeof createMemoryRouter>
) => {
  const router = createMemoryRouter(...routing);
  const rendered = render(element, {
    wrapper: () => <RouterProvider router={router} />,
  });
  return {
    router,
    ...rendered,
  };
};

export const renderWithQueryClientAndMemoryRouter = (
  element: ReactElement,
  ...routing: Parameters<typeof createMemoryRouter>
) => {
  const router = createMemoryRouter(...routing);
  const rendered = render(element, {
    wrapper: () => (
      <QueryClientProvider client={createTestQueryClient()}>
        <RouterProvider router={router} />
      </QueryClientProvider>
    ),
  });
  return {
    router,
    ...rendered,
  };
};
