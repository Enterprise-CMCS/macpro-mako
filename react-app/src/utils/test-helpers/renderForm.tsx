import { waitForElementToBeRemoved } from "@testing-library/react";
import React, { ReactElement } from "react";
import items from "mocks/data/items";
import { Authority } from "shared-types";
import { renderWithQueryClient, renderWithQueryClientAndMemoryRouter } from "./render";

export const renderFormAsync = async (form: ReactElement) => {
  const container = await renderWithQueryClient(form);

  // wait for loading
  if (container.queryAllByLabelText("three-dots-loading")?.length > 0) {
    await waitForElementToBeRemoved(() => container.queryAllByLabelText("three-dots-loading"));
  }

  return container;
};

export const renderFormWithPackageSectionAsync = async (
  form: ReactElement,
  id: string,
  authority?: string,
) => {
  const routes = [
    {
      path: "/dashboard",
      element: <h1>dashboard test</h1>,
    },
    {
      path: "/test/:id/:authority",
      element: form,
    },
  ];
  const routeOptions = {
    initialEntries: [
      `/test/${id}/${(authority || items[id]?._source?.authority || "Medicaid SPA") as Authority}`,
    ],
  };

  const container = await renderWithQueryClientAndMemoryRouter(form, routes, routeOptions);

  // wait for loading
  if (container.queryAllByLabelText("three-dots-loading")?.length > 0) {
    await waitForElementToBeRemoved(() => container.queryAllByLabelText("three-dots-loading"));
  }

  return container;
};
