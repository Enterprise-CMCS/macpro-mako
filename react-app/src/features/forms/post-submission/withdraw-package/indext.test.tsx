import { CAPITATED_AMEND_ITEM_ID } from "mocks";
import { describe, expect, it, vi } from "vitest";

import { renderFormWithPackageSectionAsync } from "@/utils/test-helpers/renderForm";

import { WithdrawPackageActionWaiver } from ".";

vi.mock("react-router", async () => ({
  ...(await vi.importActual<Record<string, unknown>>("react-router")),
  useParams: vi.fn().mockReturnValue({ authority: "1915(c)", id: CAPITATED_AMEND_ITEM_ID }),
}));
describe("WithdrawPackageAction components", () => {
  it("renders WithdrawPackageActionWaiver correctly", async () => {
    const container = renderFormWithPackageSectionAsync(<WithdrawPackageActionWaiver />);

    expect(await container).toMatchSnapshot();
  });
});
