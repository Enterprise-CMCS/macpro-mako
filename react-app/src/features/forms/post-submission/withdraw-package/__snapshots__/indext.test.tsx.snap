// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WithdrawPackageAction components > renders WithdrawPackageActionWaiver correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-screen-xl mx-auto px-4 lg:px-8"
      >
        <nav
          aria-label="breadcrumbs for spa or waiver choices"
          class="my-4"
          role="navigation"
        >
          <ul
            class="flex flex-wrap gap-1"
          >
            <li
              class="flex items-center text-sm"
            >
              <a
                class="underline text-sky-700 hover:text-sky-800"
                data-discover="true"
                href="/dashboard?tab=waivers"
              >
                Dashboard
              </a>
            </li>
            <li
              class="flex items-center text-sm"
            >
              <span>
                <svg
                  class="w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </span>
              <a
                class="underline text-sky-700 hover:text-sky-800"
                data-discover="true"
                href="/details/1915(c)/VA-2234.R11.01"
              >
                VA-2234.R11.01
              </a>
            </li>
            <li
              class="flex items-center text-sm"
            >
              <span>
                <svg
                  class="w-5 h-5"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m9 18 6-6-6-6"
                  />
                </svg>
              </span>
              <span
                aria-disabled="true"
                class="whitespace-nowrap"
              >
                Withdraw Package
              </span>
            </li>
          </ul>
        </nav>
        <form
          class="my-6 space-y-8 mx-auto justify-center flex flex-col"
        >
          <section
            class="p-4 border rounded-sm border-cardBorder"
            data-testid="detail-section"
          >
            <h1
              class="text-3xl font-semibold mb-2"
              data-testid="detail-section-title"
            >
              Withdraw 1915(c) Appendix K
            </h1>
            <hr
              class="my-6 bg-slate-200"
            />
            <div
              class="gap-8 flex flex-col"
              data-testid="detail-section-child"
            >
              <div>
                <span
                  class="text-red-600"
                >
                  *
                </span>
                 
                <em
                  class="text-neutral-500"
                >
                  Indicates a required field.
                </em>
                <div
                  class="mt-4"
                >
                  Complete this form to withdraw this 1915(c) Appendix K package. Once complete, you will not be able to resubmit this package. CMS will be notified and will use this content to review your request. If CMS needs any additional information, they will follow up by email.
                  <p
                    class="font-bold"
                  >
                    If you leave this page, you will lose your progress on this form.
                  </p>
                </div>
              </div>
              <section
                class="flex flex-col mb-8 space-y-8"
              >
                <div>
                  <p
                    class="font-bold"
                  >
                    Waiver Number
                  </p>
                  <p
                    class="text-xl"
                  >
                    VA-2234.R11.01
                  </p>
                </div>
                <div>
                  <p
                    class="font-bold"
                  >
                    Authority
                  </p>
                  <p
                    class="text-xl"
                  >
                    1915(c) Waiver
                  </p>
                </div>
              </section>
            </div>
          </section>
          <section
            class="p-4 border rounded-sm border-cardBorder"
            data-testid="attachment-section"
          >
            <h1
              class="text-3xl font-semibold mb-2"
              data-testid="attachment-section-title"
            >
              Attachments
               
            </h1>
            <hr
              class="my-6 bg-slate-200"
            />
            <div
              class="gap-8 flex flex-col"
              data-testid="attachment-section-child"
            >
              <div>
                <p
                  class="font-medium"
                >
                  Upload your supporting documentation for withdrawal or explain your need for withdrawal in the Additional Information section.
                </p>
                <br />
                <p
                  data-testid="attachments-instructions"
                >
                  Maximum file size of 80 MB per attachment.
                   
                  <span
                    class="font-bold"
                  >
                    You can add multiple files per attachment type.
                  </span>
                   Read the description for each of the attachment types on the
                   
                  <a
                    class="text-blue-900 underline"
                    data-discover="true"
                    href="/faq/withdraw-package-waiver"
                    rel="noopener noreferrer"
                    target="faq-tab"
                  >
                    FAQ Page
                  </a>
                  .
                </p>
                <br />
                <p
                  data-testid="accepted-files"
                >
                  We accept the following file formats:
                   
                  <span
                    class="font-bold"
                  >
                    .doc, .docx, .pdf, .jpg, .xlsx, and more. 
                  </span>
                   
                  <a
                    class="text-blue-900 underline"
                    data-discover="true"
                    href="/faq/acceptable-file-formats"
                    rel="noopener noreferrer"
                    target="faq-tab"
                  >
                    See the full list
                  </a>
                  .
                </p>
              </div>
              <section
                class="space-y-8"
                data-testid="attachments-section"
              >
                <div>
                  <label
                    class="text-base leading-normal peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-bold"
                    data-testid="supportingDocumentation-label"
                    for="attachments.supportingDocumentation.files"
                  >
                    Supporting Documentation
                     
                  </label>
                  <div
                    class="w-full flex items-center justify-center border border-dashed border-[#71767a] py-6 rounded-sm"
                    role="presentation"
                    tabindex="0"
                  >
                    <p>
                      Drag file here or
                       
                      <span
                        class="text-sky-700 underline hover:cursor-pointer"
                        data-testid="supportingDocumentation-click"
                      >
                        choose from folder
                      </span>
                    </p>
                    <label
                      class="sr-only"
                      for="upload-mocked-uuid-1234"
                    >
                      Drag file here or choose from folder
                    </label>
                    <input
                      accept="image/bmp,.bmp,text/csv,.csv,application/msword,.doc,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.docx,image/gif,.gif,image/jpeg,.jpeg,application/vnd.oasis.opendocument.presentation,.odp,application/vnd.oasis.opendocument.spreadsheet,.ods,application/vnd.oasis.opendocument.text,.odt,image/png,.png,application/pdf,.pdf,application/vnd.ms-powerpoint,.ppt,application/vnd.openxmlformats-officedocument.presentationml.presentation,.pptx,application/rtf,.rtf,image/tiff,.tif,text/plain,.txt,application/vnd.ms-excel,.xls,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,.xlsx"
                      data-testid="supportingDocumentation-upload"
                      id="upload-mocked-uuid-1234"
                      multiple=""
                      style="display: none;"
                      tabindex="-1"
                      type="file"
                    />
                  </div>
                </div>
              </section>
            </div>
          </section>
          <section
            class="p-4 border rounded-sm border-cardBorder"
            data-testid="additional-info"
          >
            <h1
              class="text-3xl font-semibold mb-2"
              data-testid="additional-info-title"
            >
              Additional Information
               
            </h1>
            <hr
              class="my-6 bg-slate-200"
            />
            <div
              class="gap-8 flex flex-col"
              data-testid="additional-info-child"
            >
              <div>
                <label
                  class="text-base leading-normal peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-normal"
                  data-testid="addl-info-label"
                  for="additional-info"
                >
                  Explain your need for withdrawal, or upload supporting documentation.
                </label>
                <textarea
                  aria-describedby="character-count"
                  aria-live="off"
                  aria-multiline="true"
                  class="flex min-h-[76px] w-full rounded-sm border border-black bg-transparent px-3 py-2 text-sm shadow-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 h-[200px] resize-none"
                  id="additional-info"
                  maxlength="4000"
                  name="additionalInformation"
                />
                <p
                  id=":r1:-form-item-description"
                >
                  <span
                    aria-label="character-count"
                    aria-live="polite"
                    class="text-neutral-500"
                    id="character-count"
                    tabindex="0"
                  >
                    4000 characters remaining
                  </span>
                </p>
              </div>
            </div>
          </section>
          <div
            class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none my-2 flex-row text-sm"
            role="alert"
          >
            <svg
              class="lucide lucide-info"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="12"
                cy="12"
                r="10"
              />
              <path
                d="M12 16v-4"
              />
              <path
                d="M12 8h.01"
              />
            </svg>
            <p
              class="ml-2"
            >
              Once complete, you will not be able to resubmit this package. CMS will be notified and will use this content to review your request. If CMS needs any additional information, they will follow up by email.
            </p>
            <p
              class="font-bold ml-2"
            >
              If you leave this page, you will lose your progress on this form.
            </p>
          </div>
          <section
            class="flex justify-end gap-2 p-4 ml-auto"
          >
            <button
              class="inline-flex items-center justify-center rounded-md text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-10 py-2 px-12"
              data-testid="submit-action-form"
              disabled=""
              type="button"
            >
              Submit
            </button>
            <button
              class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-primary text-primary font-bold hover:bg-primary/10 h-10 py-2 px-12"
              data-testid="cancel-action-form"
              type="reset"
            >
              Cancel
            </button>
          </section>
        </form>
        <div
          class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none mb-8 items-center flex py-8 px-14 flex-row text-sm justify-center gap-24"
          role="alert"
        >
          <p
            class="text-lg"
          >
            Do you have questions or need support?
          </p>
          <a
            data-discover="true"
            href="/faq"
            target="_blank"
          >
            <button
              class="inline-flex items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-11 rounded-md px-8 mx-4"
            >
              View FAQs
            </button>
          </a>
        </div>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-screen-xl mx-auto px-4 lg:px-8"
    >
      <nav
        aria-label="breadcrumbs for spa or waiver choices"
        class="my-4"
        role="navigation"
      >
        <ul
          class="flex flex-wrap gap-1"
        >
          <li
            class="flex items-center text-sm"
          >
            <a
              class="underline text-sky-700 hover:text-sky-800"
              data-discover="true"
              href="/dashboard?tab=waivers"
            >
              Dashboard
            </a>
          </li>
          <li
            class="flex items-center text-sm"
          >
            <span>
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </span>
            <a
              class="underline text-sky-700 hover:text-sky-800"
              data-discover="true"
              href="/details/1915(c)/VA-2234.R11.01"
            >
              VA-2234.R11.01
            </a>
          </li>
          <li
            class="flex items-center text-sm"
          >
            <span>
              <svg
                class="w-5 h-5"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m9 18 6-6-6-6"
                />
              </svg>
            </span>
            <span
              aria-disabled="true"
              class="whitespace-nowrap"
            >
              Withdraw Package
            </span>
          </li>
        </ul>
      </nav>
      <form
        class="my-6 space-y-8 mx-auto justify-center flex flex-col"
      >
        <section
          class="p-4 border rounded-sm border-cardBorder"
          data-testid="detail-section"
        >
          <h1
            class="text-3xl font-semibold mb-2"
            data-testid="detail-section-title"
          >
            Withdraw 1915(c) Appendix K
          </h1>
          <hr
            class="my-6 bg-slate-200"
          />
          <div
            class="gap-8 flex flex-col"
            data-testid="detail-section-child"
          >
            <div>
              <span
                class="text-red-600"
              >
                *
              </span>
               
              <em
                class="text-neutral-500"
              >
                Indicates a required field.
              </em>
              <div
                class="mt-4"
              >
                Complete this form to withdraw this 1915(c) Appendix K package. Once complete, you will not be able to resubmit this package. CMS will be notified and will use this content to review your request. If CMS needs any additional information, they will follow up by email.
                <p
                  class="font-bold"
                >
                  If you leave this page, you will lose your progress on this form.
                </p>
              </div>
            </div>
            <section
              class="flex flex-col mb-8 space-y-8"
            >
              <div>
                <p
                  class="font-bold"
                >
                  Waiver Number
                </p>
                <p
                  class="text-xl"
                >
                  VA-2234.R11.01
                </p>
              </div>
              <div>
                <p
                  class="font-bold"
                >
                  Authority
                </p>
                <p
                  class="text-xl"
                >
                  1915(c) Waiver
                </p>
              </div>
            </section>
          </div>
        </section>
        <section
          class="p-4 border rounded-sm border-cardBorder"
          data-testid="attachment-section"
        >
          <h1
            class="text-3xl font-semibold mb-2"
            data-testid="attachment-section-title"
          >
            Attachments
             
          </h1>
          <hr
            class="my-6 bg-slate-200"
          />
          <div
            class="gap-8 flex flex-col"
            data-testid="attachment-section-child"
          >
            <div>
              <p
                class="font-medium"
              >
                Upload your supporting documentation for withdrawal or explain your need for withdrawal in the Additional Information section.
              </p>
              <br />
              <p
                data-testid="attachments-instructions"
              >
                Maximum file size of 80 MB per attachment.
                 
                <span
                  class="font-bold"
                >
                  You can add multiple files per attachment type.
                </span>
                 Read the description for each of the attachment types on the
                 
                <a
                  class="text-blue-900 underline"
                  data-discover="true"
                  href="/faq/withdraw-package-waiver"
                  rel="noopener noreferrer"
                  target="faq-tab"
                >
                  FAQ Page
                </a>
                .
              </p>
              <br />
              <p
                data-testid="accepted-files"
              >
                We accept the following file formats:
                 
                <span
                  class="font-bold"
                >
                  .doc, .docx, .pdf, .jpg, .xlsx, and more. 
                </span>
                 
                <a
                  class="text-blue-900 underline"
                  data-discover="true"
                  href="/faq/acceptable-file-formats"
                  rel="noopener noreferrer"
                  target="faq-tab"
                >
                  See the full list
                </a>
                .
              </p>
            </div>
            <section
              class="space-y-8"
              data-testid="attachments-section"
            >
              <div>
                <label
                  class="text-base leading-normal peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-bold"
                  data-testid="supportingDocumentation-label"
                  for="attachments.supportingDocumentation.files"
                >
                  Supporting Documentation
                   
                </label>
                <div
                  class="w-full flex items-center justify-center border border-dashed border-[#71767a] py-6 rounded-sm"
                  role="presentation"
                  tabindex="0"
                >
                  <p>
                    Drag file here or
                     
                    <span
                      class="text-sky-700 underline hover:cursor-pointer"
                      data-testid="supportingDocumentation-click"
                    >
                      choose from folder
                    </span>
                  </p>
                  <label
                    class="sr-only"
                    for="upload-mocked-uuid-1234"
                  >
                    Drag file here or choose from folder
                  </label>
                  <input
                    accept="image/bmp,.bmp,text/csv,.csv,application/msword,.doc,application/vnd.openxmlformats-officedocument.wordprocessingml.document,.docx,image/gif,.gif,image/jpeg,.jpeg,application/vnd.oasis.opendocument.presentation,.odp,application/vnd.oasis.opendocument.spreadsheet,.ods,application/vnd.oasis.opendocument.text,.odt,image/png,.png,application/pdf,.pdf,application/vnd.ms-powerpoint,.ppt,application/vnd.openxmlformats-officedocument.presentationml.presentation,.pptx,application/rtf,.rtf,image/tiff,.tif,text/plain,.txt,application/vnd.ms-excel,.xls,application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,.xlsx"
                    data-testid="supportingDocumentation-upload"
                    id="upload-mocked-uuid-1234"
                    multiple=""
                    style="display: none;"
                    tabindex="-1"
                    type="file"
                  />
                </div>
              </div>
            </section>
          </div>
        </section>
        <section
          class="p-4 border rounded-sm border-cardBorder"
          data-testid="additional-info"
        >
          <h1
            class="text-3xl font-semibold mb-2"
            data-testid="additional-info-title"
          >
            Additional Information
             
          </h1>
          <hr
            class="my-6 bg-slate-200"
          />
          <div
            class="gap-8 flex flex-col"
            data-testid="additional-info-child"
          >
            <div>
              <label
                class="text-base leading-normal peer-disabled:cursor-not-allowed peer-disabled:opacity-70 font-normal"
                data-testid="addl-info-label"
                for="additional-info"
              >
                Explain your need for withdrawal, or upload supporting documentation.
              </label>
              <textarea
                aria-describedby="character-count"
                aria-live="off"
                aria-multiline="true"
                class="flex min-h-[76px] w-full rounded-sm border border-black bg-transparent px-3 py-2 text-sm shadow-sm placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 h-[200px] resize-none"
                id="additional-info"
                maxlength="4000"
                name="additionalInformation"
              />
              <p
                id=":r1:-form-item-description"
              >
                <span
                  aria-label="character-count"
                  aria-live="polite"
                  class="text-neutral-500"
                  id="character-count"
                  tabindex="0"
                >
                  4000 characters remaining
                </span>
              </p>
            </div>
          </div>
        </section>
        <div
          class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none my-2 flex-row text-sm"
          role="alert"
        >
          <svg
            class="lucide lucide-info"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <path
              d="M12 16v-4"
            />
            <path
              d="M12 8h.01"
            />
          </svg>
          <p
            class="ml-2"
          >
            Once complete, you will not be able to resubmit this package. CMS will be notified and will use this content to review your request. If CMS needs any additional information, they will follow up by email.
          </p>
          <p
            class="font-bold ml-2"
          >
            If you leave this page, you will lose your progress on this form.
          </p>
        </div>
        <section
          class="flex justify-end gap-2 p-4 ml-auto"
        >
          <button
            class="inline-flex items-center justify-center rounded-md text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-10 py-2 px-12"
            data-testid="submit-action-form"
            disabled=""
            type="button"
          >
            Submit
          </button>
          <button
            class="inline-flex items-center justify-center rounded-md text-[16px] ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-primary text-primary font-bold hover:bg-primary/10 h-10 py-2 px-12"
            data-testid="cancel-action-form"
            type="reset"
          >
            Cancel
          </button>
        </section>
      </form>
      <div
        class="relative w-full border p-4 [&>svg~*]:pl-7 [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-4 [&>svg]:text-foreground border-l-[6px] border-y-0 border-r-0 border-cyan-500 bg-cyan-300/10 rounded-none mb-8 items-center flex py-8 px-14 flex-row text-sm justify-center gap-24"
        role="alert"
      >
        <p
          class="text-lg"
        >
          Do you have questions or need support?
        </p>
        <a
          data-discover="true"
          href="/faq"
          target="_blank"
        >
          <button
            class="inline-flex items-center justify-center text-[16px] font-bold ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-slate-50 hover:bg-primary-dark h-11 rounded-md px-8 mx-4"
          >
            View FAQs
          </button>
        </a>
      </div>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
